RecoParticleFlow/PFClusterProducer/plugins/SimMappers/ComputeClusterTime.h

```diff
@@ -0,0 +1,133 @@
+#ifndef ComputeClusterTime_h
+#define ComputeClusterTime_h
+
+// user include files
+#include <vector>
+
+
+// functions to select the hits to compute the time of a given cluster
+// start with the only hits with timing information
+// average among the hits contained in the chosen time interval
+
+// N.B. time is corrected wrt vtx-calorimeter distance 
+// with straight line and light speed hypothesis
+// for charged tracks or heavy particles (longer track length or beta < 1) 
+// need to correct the offset at analysis level
+
+
+
+namespace hgcalSimClusterTime {
+
+  //time-interval based on that ~210ps wide and with the highest number of hits
+  float fixSizeHighestDensity(std::vector<float>& t, float deltaT=0.210 /*time window in ns*/, float timeWidthBy=0.5){
+
+    float tolerance = 0.05f;
+    std::sort(t.begin(), t.end());
+
+    int max_elements = 0;
+    int start_el = 0;
+    int end_el = 0;
+    float timeW = 0.f;
+
+    for(auto start = t.begin(); start != t.end(); ++start) {
+      const auto startRef = *start;
+      int c = count_if(start, t.end(), [&](float el) {
+	  return el - startRef <= deltaT + tolerance;
+	});
+      if (c > max_elements) {
+	max_elements = c;
```

Comment:
tabs and spaces for indentation are still mixed throughout this file and the next one.